tests/std/tests/P0401R6_allocate_at_least/test.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+#include <numeric>
+#include <type_traits>
+
+using namespace std;
+
+static_assert(is_standard_layout_v<allocation_result<int*>>);
+static_assert(is_trivially_copyable_v<allocation_result<int*>>);
+
+struct generous_allocator {
+    using value_type = int;
+
+    allocator<int> inner;
+
+    [[nodiscard]] _CONSTEXPR20_DYNALLOC int* allocate(const size_t count) {
+        return inner.allocate(max(count, static_cast<size_t>(16)));
+    }
+
+    [[nodiscard]] _CONSTEXPR20_DYNALLOC allocation_result<int*> allocate_at_least(const size_t count) {
+        return {inner.allocate(max(count, static_cast<size_t>(16))), max(count, static_cast<size_t>(16))};
+    }
+
+    _CONSTEXPR20_DYNALLOC void deallocate(int* ptr, const size_t count) {
+        inner.deallocate(ptr, max(count, static_cast<size_t>(16)));
+    }
+};
+
+struct strict_allocator {
+    using value_type = int;
+
+    allocator<int> inner;
+
+    [[nodiscard]] _CONSTEXPR20_DYNALLOC int* allocate(const size_t count) {
+        return inner.allocate(count);
+    }
+
+    _CONSTEXPR20_DYNALLOC void deallocate(int* ptr, const size_t count) {
+        inner.deallocate(ptr, count);
+    }
+};
+
+_CONSTEXPR20_DYNALLOC bool test() {

```

Comment:
I would like some tests for the concept, to ensure that it works as expected especially with some badly handcrafted allocators

Reply:
Currently we test 2 valid options:
1. allocator that has `allocate_at_least` and it satisfies requirements
2. allocator that doesn't have `allocate_at_least`

There aren't more valid options so there is nothing more to test.